toml/src/test/kotlin/org/rust/toml/search/CargoTomlFindUsagesTest.kt

```diff
@@ -0,0 +1,132 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.toml.search
+
+import com.intellij.codeInsight.TargetElementUtil
+import com.intellij.lang.LanguageCommenters
+import com.intellij.openapi.vfs.VfsUtil
+import com.intellij.openapi.vfs.VfsUtilCore
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiNamedElement
+import org.rust.cargo.RsWithToolchainTestBase
+import org.rust.cargo.project.model.cargoProjects
+import org.rust.lang.core.psi.ext.startOffset
+import org.rust.openapiext.document
+import org.rust.openapiext.toPsiFile
+
+class CargoTomlFindUsagesTest : RsWithToolchainTestBase() {
+
+    fun test() = buildProject {
+        toml("Cargo.toml", """
+            [package]
+            name = "hello"
+            version = "0.1.0"
+            authors = []
+
+            [features]
+            foo = ["dep_pkg/feature_bar"] # - Cargo feature dependency
+
+            [dependencies]
+            dep_pkg = { path = "./dep_pkg", features = ["feature_bar"] } # - Package dependency
+        """)
+
+        dir("src") {
+            rust("main.rs", """
+                fn main() {}
+            """)
+        }
+
+        dir("dep_pkg") {
+            toml("Cargo.toml", """
+                [package]
+                name = "dep_pkg"
+                version = "0.1.0"
+                authors = []
+
+                [features]
+                feature_foo = ["feature_bar"] # - Cargo feature dependency
+                feature_bar = []
+                #^
+            """)
+
+            dir("src") {
+                rust("lib.rs", """
+                    #[cfg(feature = "feature_bar")] // - Cfg attribute
+                    #[cfg_attr(feature = "feature_bar", allow(all))] // - Cfg attribute
+                    fn foobar() {}
+                """)
+            }
+        }
+    }.run {
+        assertNotNull(project.cargoProjects.allProjects.single().workspace)
```

Comment:
Maybe it's worth to extract code from `run` into a separate method? Currently, it's quite huge and cannot be reused if needed